modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/resource/v1_0/test/ObjectEntryResourceTest.java

```diff
@@ -7302,6 +7388,74 @@ public void testGetObjectRelationshipERCFieldNameInOneToManyRelationshipFromRela
 			_objectEntry1.getExternalReferenceCode());
 	}
 
+	@Test
+	public void testGetScopeScopeKeyByExternalReferenceCode() throws Exception {
+		_siteScopedObjectEntry1 = ObjectEntryTestUtil.addObjectEntry(
+			_siteScopedObjectDefinition1, _OBJECT_FIELD_NAME_1,
+			_OBJECT_FIELD_VALUE_1);
+
+		Group group = _groupLocalService.fetchGroup(
+			TestPropsValues.getGroupId());
+
+		// Using groupExternalReferenceCode
+
+		JSONObject jsonObject = HTTPTestUtil.invokeToJSONObject(
+			null,
+			StringBundler.concat(
+				_getEndpoint(
+					group.getExternalReferenceCode(), 0,
+					_siteScopedObjectDefinition1),
+				"/by-external-reference-code/",
+				_siteScopedObjectEntry1.getExternalReferenceCode()),
+			Http.Method.GET);
+
+		Assert.assertEquals(
+			String.valueOf(_OBJECT_FIELD_VALUE_1),
+			jsonObject.get(_OBJECT_FIELD_NAME_1));
+
+		Assert.assertEquals(
+			jsonObject.getLong("id"),
+			_siteScopedObjectEntry1.getObjectEntryId());
```

Comment:
Do we need these separate asserts or could we inline/merge the  HTTPTestUtil.invokeToJSONObject() +Assert (comparing jsons)?

Reply:
consider this comment to the following asserts in this method :+1: